test/test_autograd.py

```diff
@@ -6112,6 +6112,18 @@ def _test_euclidean_large_cdist(sizex, sizey=None):
         _test_cdist_for_size((1, 1), (S, 1))
         _test_euclidean_large_cdist((2000, 5))
 
+    # Ensure that cdist backward with p<1 does not produce NaNs
+    def test_cdist_grad_p_lt_1_no_nan(self, device):
+        for p in [0.99, 0.7, 0.5, 0.1, 0.01]:
+            x = torch.randn(1, 4, device=device)
+            y = x + torch.tensor([[1.0, 0.0, 1e-45, 1e-46]], device=device)
```

Comment:
Not sure if it's really useful to test 1e-45/1e-46.
If you just want to check the value at 0 that should be fine.

Note that if you actually use a small value that is not 0 (like 1e-6), you will most likely still get numerical issues (like we do for other similar ops: https://github.com/pytorch/pytorch/issues/43211).

Reply:
Yeah, good point. I've removed 1e-45 and 1e-46.